src/Microsoft.ML.Transforms/Expression/BuiltinFunctions.cs

```diff
@@ -0,0 +1,1095 @@
+            for (int i = 0; i < values.Length; i++)
+            {
+                if (FunctionProviderUtils.IsNA(values[i]))
+                {
+                    Contracts.Assert(values.Length > 1);
+                    return FunctionProviderUtils.GetNA(fn.ReturnType);
+                }
+            }
+
+            return null;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static R8 Pi()
+        {
+            return Math.PI;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static R4 NA(R4 a)
+        {
+            return R4.NaN;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static R8 NA(R8 a)
+        {
+            return R8.NaN;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static I4 Default(I4 a)
+        {
+            return default(I4);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static I8 Default(I8 a)
+        {
+            return default(I8);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static R4 Default(R4 a)
+        {
+            return default(R4);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static R8 Default(R8 a)
+        {
+            return default(R8);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static BL Default(BL a)
+        {
+            return default(BL);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static TX Default(TX a)
+        {
+            return default(TX);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static R4 Sign(R4 a)
+        {
+            // Preserves NaN. Unfortunately, it also preserves negative zero,
+            // but perhaps that is a good thing?
+            return a > 0 ? +1 : a < 0 ? -1 : a;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static R8 Sign(R8 a)
+        {
+            // Preserves NaN. Unfortunately, it also preserves negative zero,
+            // but perhaps that is a good thing?
+            return a > 0 ? +1 : a < 0 ? -1 : a;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static I4 Sign(I4 a)
+        {
+            return a > 0 ? +1 : a < 0 ? -1 : a;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static I8 Sign(I8 a)
+        {
+            return a > 0 ? +1 : a < 0 ? -1 : a;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static R4 Pow(R4 a, R4 b)
+        {
+            return (R4)Math.Pow(a, b);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static R4 Exp(R4 a)
+        {
+            return (R4)Math.Exp(a);
```

Comment:
How difficult would it be to output an ONNX graph instead of (or in addition to) building up .NET?

Background: 
The AutoML team is looking to support basic math functionality like: `LN(x)`, `x-y`, and `EXP(x)`. They could use the expression transform, though they need ONNX support. I'm hoping to avoid creating three new transforms for these operations by using the expression transform. This will avoid an endlessly growling list of tiny, single function, math transforms.

Reply:
Issue tracked: https://github.com/dotnet/machinelearning/issues/4615